index.js

```diff
@@ -90,6 +90,7 @@ class YoutubeVideoElement extends HTMLElement {
     this.shadowRoot.appendChild(template.content.cloneNode(true));
 
     const src = this.getAttribute('src');
+    this.onPlayerReadyQueue = [];
```

Comment:
The `ytReadyQueue` above doesn't work for things like calling `this.ytPlayer.setVolume()`  or `this.ytPlayer.play()`. We actually need a different queue that waits until our instance of `YT.Player` is ready

Reply:
That makes sense but there might still be an issue here if the iframe isn't loaded yet when the API is called. I think we need to wait for both ytReady and iframe.onload. Not sure if we can guarantee the order of those either.

Reply:
I didn't go through the YT API script but I imagine the YT `onReady` event is fired when the API can accept calls and it might be even before that, right after an instance of `YT.Player` is made. That's how the player.js Vimeo API works at least. not sure if waiting on `iframe.onload` is needed here, wouldn't that wait until really all assets are loaded in the iframe.